packages/core/certora/optimistic-asserter/specs/Asserter_Auxiliary.spec

```diff
@@ -0,0 +1,186 @@
+// For every new spec you create, import this basic spec file:
+import "./Asserter_Base.spec"
+import "./dispatchedMethods.spec"
+
+using TestnetERC20 as testERC20
+/**************************************************
+ *      Top Level Properties / Rule Ideas         *
+ **************************************************/
+// Here you can write your ideas for rules and implement them 
+// in this spec.
+
+/**************************************************
+ *                  Misc. rules                   *
+ **************************************************/
+
+// A simple rule that checks which of the main contract methods
+// are reachable (reach the assert false statement after function call).
+ rule sanity(method f) {
+    env e;  // Environment variable - includes all transaction and block information.
+    calldataarg args; // arbitrary calldata - adapted for every method signature implicitly
+    f(e, args);
+    assert false;
```

Comment:
Just to make sure I understand (I think we already discussed), this will _fail_ for all reachable methods, right?

Is it done this way because there is no way to have it fail for methods that are unreachable?

Reply:
Right, so this is basically just a detector for unreachable functions -- it just counterintuitively (due to the constraints of the system) passes for methods that are unreachable (and therefore are problematic).